src/Tizen.Account.OAuth2/Tizen.Account.OAuth2/Authorizer.cs

```diff
@@ -98,83 +103,98 @@ private TokenResponse GetRefreshToken(IntPtr requestHandle)
-                    }
-
-                    long expiresIn;
-                    ret = Interop.Response.GetExpiresIn(responseHandle, out expiresIn);
-                    if (ret != (int)OAuth2Error.None)
-                    {
-                        Log.Error(ErrorFactory.LogTag, "Failed to get expires in");
-                    }
-
-                    IntPtr refreshToken;
-                    ret = Interop.Response.GetRefreshToken(responseHandle, out refreshToken);
-                    if (ret != (int)OAuth2Error.None)
-                    {
-                        Log.Error(ErrorFactory.LogTag, "Interop failed");
-                        throw ErrorFactory.GetException(ret);
-                    }
-
-                    IntPtr scope;
-                    ret = Interop.Response.GetScope(responseHandle, out scope);
-                    if (ret != (int)OAuth2Error.None)
-                    {
-                        Log.Error(ErrorFactory.LogTag, "Failed to get scope");
-                    }
+                    Log.Error(ErrorFactory.LogTag, "Interop failed");
+                    throw ErrorFactory.GetException(ret);
+                }
+            }
+            tcsaccessTokenResponse.Task.ConfigureAwait(true);
+            Interop.Request.Destroy(requestHandle);
+            return tcsaccessTokenResponse.Task.Result;
+        }
 
-                    IEnumerable<string> scopes = (scope == IntPtr.Zero) ? null : Marshal.PtrToStringAnsi(scope)?.Split(' ');
+        private static void AccessTokenCb(IntPtr responseHandle, IntPtr usrData)
+        {
+            int ret = (int)OAuth2Error.None;
+            IntPtr error = IntPtr.Zero;
+            TokenResponse response = null;
+            IntPtr responseCompletionId = usrData;
+            TaskCompletionSource <TokenResponse> responseCompletionSource = _taskResponseMap[responseCompletionId];
+            _taskResponseMap.Remove(responseCompletionId);
 
-                    var token = new AccessToken();
-                    token.Token = (accessToken == IntPtr.Zero) ? null : Marshal.PtrToStringAnsi(accessToken);
-                    token.TokenType = (tokenType == IntPtr.Zero) ? null : Marshal.PtrToStringAnsi(tokenType);
-                    token.Scope = scopes;
-                    token.ExpiresIn = expiresIn;
+            Interop.Response.GetError(responseHandle, out error);
+            if (error != IntPtr.Zero)
+            {
+                Log.Error(ErrorFactory.LogTag, "Error occured");
+                responseCompletionSource.SetException(ErrorFactory.GetException(error));
+            }
+            else
+            {
+                IntPtr accessToken;
+                ret = Interop.Response.GetAccessToken(responseHandle, out accessToken);
+                if (ret != (int)OAuth2Error.None)
+                {
+                    Log.Error(ErrorFactory.LogTag, "Interop failed");
+                    responseCompletionSource.SetException(ErrorFactory.GetException(ret));
```

Comment:
```suggestion
                    responseCompletionSource.SetException(ErrorFactory.GetException(ret));
                    return;
```
You should return here

Reply:
Implemented